frontend/src/libs/components/prompt-generation/libs/components/prompt-card/prompt-card.tsx

```diff
@@ -0,0 +1,50 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useEffect, useState } from '~/libs/hooks/hooks.js';
+
+import { type PromptCategoryValue } from '../../enums/enums.js';
+import { categoryPrompts } from '../../mocks/mocks.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  category: PromptCategoryValue;
+  isSpinning: boolean;
+};
+
+const PromptCard: React.FC<Properties> = ({ category, isSpinning }) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const prompts = categoryPrompts[category];
+
+  useEffect(() => {
+    if (isSpinning) {
+      const randomIndex = Math.floor(Math.random() * prompts.length);
+      setCurrentIndex(randomIndex);
+    }
+  }, [isSpinning, prompts]);
```

Comment:
Not sure if we need spinning on category prompt card - as per their generation me may just call the backend to generate a random prompt - so don't know if it may work with single request without mocked data

Reply:
The array of prompts is required to scroll through. So, what should I do for now? I can remove the animation, and just show generatedPrompt once it is generated.

Reply:
Yep